### ISSUE TITLE
Configuration parameter gets duplicated ( several times)

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#988]
In GitLab by @orioltinto on Mar 23, 2023, 10:49

Hello again @dbeltrankyl and @kinow,

### Autosubmit Version

4.0.75

### Summary

I have defined some parameters in a configuration file like:

```yaml
python_environment:
  # Name of the virtual environment in the remote platform experiment folder
  folder_name: python_environment
  requirements:
    # Because there's an issue with numba, for now we need to keep a specific version of numpy
    - numpy==1.23
    - enstools-compression
    # Just to try a library from a git repository.
    - git+https://gitlab.physik.uni-muenchen.de/Oriol.Tinto/otils.git
    - f90nml

```

When I do autosubmit report I get:

```

PYTHON_ENVIRONMENT.REQUIREMENTS=['numpy==1.23', 'enstools-compression', 'git+https://gitlab.physik.uni-muenchen.de/Oriol.Tinto/otils.git', 'f90nml', 'numpy==1.23', 'enstools-compression', 'git+https://gitlab.physik.uni-muenchen.de/Oriol.Tinto/otils.git', 'f90nml', 'numpy==1.23', 'enstools-compression', 'git+https://gitlab.physik.uni-muenchen.de/Oriol.Tinto/otils.git', 'f90nml', 'numpy==1.23', 'enstools-compression', 'git+https://gitlab.physik.uni-muenchen.de/Oriol.Tinto/otils.git', 'f90nml', 'numpy==1.23', 'enstools-compression', 'git+https://gitlab.physik.uni-muenchen.de/Oriol.Tinto/otils.git', 'f90nml', 'numpy==1.23', 'enstools-compression', ...
```

The same things repeated up to 195 times.

### Steps to reproduce

I guess you can reproduce the problem creating an experiment with a similar configuration and doing autosubmit report.

()

**Comment by kinow** [comment]
In GitLab by @orioltinto on Mar 23, 2023, 14:20

Hi @dbeltrankyl,

The thing is that this was working as what I was expecting in 4.0.71

It looks to me like a recursion issue somewhere, might it be related with the change you did to enable using folders again?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 23, 2023, 14:24

Well, in 4.0.71 we had an older configuration parser version that supported this list. 

However, introducing the new distributed config breaks this compatibility as the postprocessing that is done is not yet compatible with YML list syntax. 

The function that is messing it is the unify_config together with the dynamic_variables replacement. I fixed the custom_config section, but it is not yet general.

I'll try to have it solved next week.

Edit: I had a change of mind, as I'm testing master branch right now,  I'll see if it would be hard to include in the next tag )

**Comment by kinow** [comment]
In GitLab by @mcastril on Mar 23, 2023, 14:53

> Edit: I had a change of mind, as I'm testing master branch right now, I'll see if it would be hard to include in the next tag )

You mean the next Autosubmit 4.0 tag right? I think it's better to "fix" 4.0 to not introduce a recursion like Oriol said

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 23, 2023, 14:53

Yes, in next beta tag I mean, before I install it to LUMI VM

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 23, 2023, 16:29

Found the bug. Actually, this append thing would also happen with the old distributed conf when you try to override lists. 

It should be solved in the next beta tag. I'll update the issue when it is ready.


The report command in my local branch now reports: 

`AAA_TEST.TEST=['test', 'test2', 'test', 'working', 'duplicating']`

I have one doubt tho, 

What would be the best approach when merging lists...

a) [ Implemented one ] Override the whole list for the new list. If the contents are different, this would be the consistent one if I were to compare it with the other types 

b) Append the new values that aren't in the current list ?

c) Give an option like %AUTOSUBMIT.LIST_OVERRIDE_BEHAVIOUR% Default = a) ? 

I would go for a) but what do you think? @mcastril

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 23, 2023, 16:31

mentioned in commit cb669f8cc2be3e612e6641a8850d3d692441b56a

**Comment by kinow** [comment]
In GitLab by @mcastril on Mar 23, 2023, 16:45

I also think it should be a) unless there's a very strong reason. Otherwise, it would create an exception wrt other overrides work.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 30, 2023, 09:56

This should be working fine in the last pip/tag

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Apr 27, 2023, 09:21

Fixed in v4.0.0b13

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Apr 27, 2023, 09:21

unassigned @dbeltrankyl
